Libplanet/Blockchain/Renderers/IRenderer.cs

```diff
@@ -0,0 +1,132 @@
+        /// <param name="context">The equivalent context object to an object passed to
+        /// the <paramref name="action"/>'s <see cref="IAction.Execute(IActionContext)"/> method.
+        /// That means <see cref="IActionContext.PreviousStates"/> are the states right
+        /// <em>before</em> this action executed.  For the states after this action executed,
+        /// use the <paramref name="nextStates"/> argument instead.</param>
+        /// <param name="nextStates">The states right <em>after</em> this action executed,
+        /// which means it is equivalent to the states <paramref name="action"/>'s
+        /// <see cref="IAction.Execute(IActionContext)"/> method returned.</param>
+        /// <remarks>As a rule of thumb, this should be the inverse of
+        /// <see cref="RenderAction(IAction, IActionContext, IAccountStateDelta)"/> method
+        /// with redrawing the graphics on the display at the finish.</remarks>
+        /// <remarks>The reason why the parameter <paramref name="action"/> takes
+        /// <see cref="IAction"/> instead of <typeparamref name="T"/> is because it can take
+        /// block actions (<see cref="Policies.IBlockPolicy{T}.BlockAction"/>) besides transaction
+        /// actions (<see cref="Tx.Transaction{T}.Actions"/>).</remarks>
+        void UnrenderAction(IAction action, IActionContext context, IAccountStateDelta nextStates);
+
+        /// <summary>
+        /// Does the similar things to <see cref=
+        /// "RenderAction(IAction, IActionContext, IAccountStateDelta)"/>, except that this method
+        /// is invoked when <paramref name="action"/> has terminated with an exception.
+        /// </summary>
+        /// <param name="action">An action which threw an exception during execution.</param>
+        /// <param name="context">The equivalent context object to an object passed to
+        /// the <paramref name="action"/>'s <see cref="IAction.Execute(IActionContext)"/> method.
+        /// That means <see cref="IActionContext.PreviousStates"/> are the states right
+        /// <em>before</em> this action executed.</param>
+        /// <param name="exception">The exception thrown during executing the <paramref
+        /// name="action"/>.</param>
+        /// <remarks>The reason why the parameter <paramref name="action"/> takes
+        /// <see cref="IAction"/> instead of <typeparamref name="T"/> is because it can take
+        /// block actions (<see cref="Policies.IBlockPolicy{T}.BlockAction"/>) besides transaction
+        /// actions (<see cref="Tx.Transaction{T}.Actions"/>).</remarks>
+        void RenderActionError(IAction action, IActionContext context, Exception exception);
+
+        /// <summary>
+        /// Does the similar things to <see
+        /// cref="UnrenderAction(IAction, IActionContext, IAccountStateDelta)"/>, except that
+        /// this methodis invoked when <paramref name="action"/> has terminated with an exception.
```

Comment:
```suggestion
        /// this methods invoked when <paramref name="action"/> has terminated with an exception.
```

Reply:
I wanted to say “this method is”… Anyway I amended!